wallet.js

```diff
@@ -0,0 +1,359 @@
+    const { keys } = newWalletData;
+
+    const updatedHistoryMap = {};
+    const updatedBalanceMap = {};
+    const balances = this.getTxBalance(newTx);
+
+    // we now loop through all tokens present in the new tx to get the new history and balance
+    for (const [tokenUid, tokenTxBalance] of Object.entries(balances)) {
+      // we may not have this token yet, so state.tokensHistory[tokenUid] would return undefined
+      const currentHistory = state.tokensHistory[tokenUid] || [];
+      const newTokenHistory = addTxToSortedList(tokenUid, tx, tokenTxBalance, currentHistory);
+      updatedHistoryMap[tokenUid] = newTokenHistory;
+      // totalBalance should not be confused with tokenTxBalance. The latter is the balance of the new
+      // tx, while the former is the total balance of the token, considering all tx history
+      const totalBalance = getBalance(tokenUid);
+      updatedBalanceMap[tokenUid] = totalBalance;
+    }
+    const newTokensHistory = Object.assign({}, state.tokensHistory, updatedHistoryMap);
+    const newTokensBalance = Object.assign({}, state.tokensBalance, updatedBalanceMap);
+
+    hathorLib.storage.setItem('local:tokens:history', newTokensHistory);
+    hathorLib.storage.setItem('local:tokens:balance', newTokensBalance);
+    */
+  };
+
+  /**
+   * Send a transaction with multi tokens
+   * @param {Object} data Array of {'address', 'value', 'token'}
+   */
+  sendMultiTokenTransaction(data) {
+    const txData = {
+      inputs: [],
+      outputs: []
+    }
+
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+
+    // First I need an array with all tokens
+    const allTokens = [];
+    for (const d of data) {
+      allTokens.push(d.token);
+    }
+
+    for (const d of data) {
+      const dataOutput = {'address': d.address, 'value': d.value, 'tokenData': hathorLib.tokens.getTokenIndex(allTokens, d.token.uid)};
+      const partialData = {'inputs': [], 'outputs': [dataOutput]}
+      const result = hathorLib.wallet.prepareSendTokensData(partialData, d.token, true, historyTxs, allTokens);
+
+      if (!result.success) {
+        console.log('Error sending tx:', result.message);
+        return;
+      }
+
+      const dataToken = result.data;
+
+      txData['inputs'] = [...txData['inputs'], ...dataToken['inputs']];
+      txData['outputs'] = [...txData['outputs'], ...dataToken['outputs']];
+    }
+
+    txData.tokens = hathorLib.tokens.filterTokens(allTokens, hathorLib.constants.HATHOR_TOKEN_CONFIG).map((token) => token.uid);
+
+    const updateOutputs = (val) => {
+      const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+      for (const input of txData.inputs) {
+        historyTxs[input.tx_id]['outputs'][input.index]['spent_by'] = val;
+      }
+    }
+
+    // XXX Setting input as spent by, so it won't be chosen again
+    updateOutputs('2');
```

Comment:
I know it's a PoC and we could continue with this here in this PR but we should handle this properly soon.

Reply:
You're right. We must also update the input status when a transaction is updated.

Reply:
Yes, but again...for a PoC I am ok going with it

Reply:
I marked it as a TODO item. So, we can review it when adding it to the hathor-wallet-lib. What do think? Can I mark it as solved?

Reply:
Ok for me.